opencti-platform/opencti-graphql/src/http/httpPlatform.js

```diff
@@ -458,8 +458,30 @@ const createApp = async (app, schema) => {
   // -- Passport callback
   // -- Default limit is '100kb' based on https://expressjs.com/en/resources/middleware/body-parser.html
   const urlencodedParser = AUTH_PAYLOAD_BODY_SIZE ? bodyParser.urlencoded({ extended: true, limit: AUTH_PAYLOAD_BODY_SIZE }) : bodyParser.urlencoded({ extended: true });
+  const sanitizeReferer = (refererToSanitize, baseUrl) => {
+    if (!refererToSanitize) return '/';
+    const base = new URL(baseUrl);
+    const refererUrl = new URL(refererToSanitize, base);
+
+    const isSameOrigin = refererUrl.origin === base.origin;
+    const isUrlRelative = refererToSanitize.startsWith('/');
+
+    if (isSameOrigin || isUrlRelative) {
+      return refererUrl.pathname + refererUrl.search + refererUrl.hash;
+    }
+    logApp.info('Error auth provider callback : url has been altered', { url: refererToSanitize });
+    return '/';
+  };
+
   app.all(`${basePath}/auth/:provider/callback`, urlencodedParser, async (req, res, next) => {
-    const referer = req.body.RelayState ?? req.session.referer;
+    const baseUrl = getBaseUrl();
+    let referer = '/';
+
+    if (req.body.RelayState) {
+      referer = sanitizeReferer(req.body.RelayState, baseUrl);
```

Comment:
I'm not sure we need to have baseUrl as argument here since there is `getBaseUrl()` function.